internal/pkg/config/service.go

```diff
@@ -0,0 +1,120 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ssm"
+)
+
+// Service represents a deployable service or task.
+type Service struct {
+	App  string `json:"App"`  // Name of the app this service belongs to.
+	Name string `json:"name"` // Name of the service, which must be unique within a app.
+	Type string `json:"type"` // Type of the service/component (LoadBalanced app, etc)
+}
+
+// CreateService instantiates a new service within an existing application. Skip if
+// the service already exists in the application.
+func (s *Store) CreateService(svc *Service) error {
+	if _, err := s.GetApplication(svc.App); err != nil {
+		return err
+	}
+
+	servicePath := fmt.Sprintf(fmtSvcParamPath, svc.App, svc.Name)
+	data, err := marshal(svc)
+	if err != nil {
+		return fmt.Errorf("serializing service %s: %w", svc.Name, err)
+	}
+
+	_, err = s.ssmClient.PutParameter(&ssm.PutParameterInput{
+		Name:        aws.String(servicePath),
+		Description: aws.String(fmt.Sprintf("Copilot Service %s", svc.Name)),
+		Type:        aws.String(ssm.ParameterTypeString),
+		Value:       aws.String(data),
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case ssm.ErrCodeParameterAlreadyExists:
+				return nil
+			}
+		}
+		return fmt.Errorf("create service %s in application %s: %w", svc.Name, svc.App, err)
+	}
+	return nil
+}
+
+// GetService gets a service belonging to a particular application by name. If no svc is found
+// it returns ErrNoSuchService.
+func (s *Store) GetService(appName, svcName string) (*Service, error) {
+	svcPath := fmt.Sprintf(fmtSvcParamPath, appName, svcName)
+	svcParam, err := s.ssmClient.GetParameter(&ssm.GetParameterInput{
+		Name: aws.String(svcPath),
+	})
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case ssm.ErrCodeParameterNotFound:
+				return nil, &ErrNoSuchService{
+					ApplicationName: appName,
+					ServiceName:     svcName,
+				}
+			}
+		}
+		return nil, fmt.Errorf("get service %s in application %s: %w", svcName, appName, err)
+	}
+
+	var svc Service
+	err = json.Unmarshal([]byte(*svcParam.Parameter.Value), &svc)
+	if err != nil {
+		return nil, fmt.Errorf("read details for service %s in application %s: %w", svcName, appName, err)
```

Comment:
should we use the word "configuration" instead of "details" in the error messages?

Reply:
:+1: